configure.ac

```diff
@@ -202,6 +202,67 @@ AS_CASE([$with_gc],
     )
 AC_MSG_RESULT([$with_gc])
 
+dnl
+dnl User setting: native thread-local storage (off by default)
+dnl
+
+AC_ARG_WITH([native-tls],
+    [AS_HELP_STRING([--with-native-tls],
+        [use native thread-local storage implementation])],
+        [],
+        [with_native_tls=no])
+AC_MSG_CHECKING([whether to use native tls])
+AS_IF([test "x$with_native_tls" == "xyes"], [
+        AC_COMPILE_IFELSE(
+            [AC_LANG_SOURCE([[
+#if defined(__APPLE__) && defined(__x86_64__)
+#error FAIL
+#endif
+        ]])],
+            [
+                AC_DEFINE([USE_NATIVE_TLS], [1], [define as 1 if using native TLS])
+                AC_MSG_RESULT([yes])
+            ],
+            [
+                AC_DEFINE([USE_PTHREAD_TLS], [1], [define as 1 if using pthread_getspecific])
+            ]
+        )
+        AC_RUN_IFELSE(
+            [AC_LANG_PROGRAM([[
+#include <pthread.h>
+#include <string.h>
+]],[[
+#if defined(__APPLE__) && defined(__x86_64__)
+    /* This is an idea borrowed from Mono. We test if the implementation
+     * of pthread_getspecific() uses the assembly code below. If that is
+     * true, we can replace calls to pthread_getspecific() with the
+     * matching inline assembly, allowing a significant performance boost.
+     */
+    static char asm_code[] = {
+        // movq %gs:(,%rdi,8), %rax
+        // retq
+        0x65, 0x48, 0x8b, 0x04, 0xfd, 0x00, 0x00, 0x00, 0x00, 0xc3
+    };
+    if (memcmp((void *)pthread_getspecific, asm_code, sizeof(asm_code)) != 0) {
+        return 1;
+    }
+    return 0;
+#else
+#error FAIL
+#endif
+            ]])],
+            [AC_DEFINE([USE_MACOS_PTHREAD_TLS_ASM], [1],
+                [define as 1 if macOS assembly implementation supported])
+            AC_MSG_RESULT([[yes (macOS assembly)]])
+            ],
+            [AC_MSG_RESULT([[yes (pthread-based)]])],
+            [AC_MSG_NOTICE([[yes (pthread-based)]])])
```

Comment:
This is confusing to read, perhaps partially because of the deep nesting... but why is there both a AC_MSG_RESULT  and a AC_MSG_NOTICE here?

Reply:
I think auto-complete tripped me up, should indeed be `AC_MSG_RESULT()` also.